climpred/prediction.py

```diff
@@ -349,7 +377,12 @@ def compute_persistence(
             # room for lead from current forecast
             a, _ = reduce_time_series(hind, reference, lag)
             inits = a['time']
-        ref = reference.sel(time=inits + lag)
+        offset_args_tuple = get_lead_cftime_shift_args(
+            getattr(hind['lead'], 'units'), lag
+        )
+        target_dates = inits.to_index().shift(*offset_args_tuple)
```

Comment:
I see this a lot. Maybe define as a util function and write a doc string so it's clear that it does?